torchinductor/ir.py

```diff
@@ -1497,6 +1498,106 @@ def _apply_loop_reordering(index_vars, sizes, memory_addrs):
         sizes = [sizes[i] for i in order]
         return sizes, inverse_reorder(order)
 
+    @staticmethod
+    def _apply_loop_reordering_channel_last(index_vars, sizes):
+        """
+        apply channel-last loop order
+        """
+        # nchw -> nhwc
+        if len(sizes) == 4:
+            order = [0, 2, 3, 1]
+            sizes = [sizes[i] for i in order]
+            index_vars = [index_vars[i] for i in order]
+            return sizes, inverse_reorder(order), index_vars
+        else:
+            return sizes, inverse_reorder(range(len(sizes))), index_vars
+
+    def reorder_channel_last(self):
```

Comment:
We should refactor this to share code rather than copying.

Reply:
Refactor simplify_reorder_and_tile with channel_last option